src/engine/guidance/post_processing.cpp

```diff
@@ -520,9 +520,18 @@ std::vector<RouteStep> anticipateLaneChangeForRoundabouts(std::vector<RouteStep>
         // FIXME: assumes right-side driving (counter-clockwise roundabout flow)
         const auto enter_direction = enter.maneuver.instruction.direction_modifier;
 
-        if (util::guidance::isRightTurn(enter.maneuver.instruction))
-            enter.maneuver.instruction.direction_modifier =
-                mirrorDirectionModifier(enter_direction);
+        if (facade.UseLeftSideDriving())
+        {
+            if (util::guidance::isLeftTurn(enter.maneuver.instruction))
+                enter.maneuver.instruction.direction_modifier =
+                    mirrorDirectionModifier(enter_direction);
+        }
+        else
+        {
+            if (util::guidance::isRightTurn(enter.maneuver.instruction))
+                enter.maneuver.instruction.direction_modifier =
+                    mirrorDirectionModifier(enter_direction);
+        }
```

Comment:
In order to test this it would be great to have clockwise-flowing roundabouts modeled as Cucumber scenarios, you can take this one as example which assumes right-sided driving (and therefore counter-clockwise roundabout flow): https://github.com/Project-OSRM/osrm-backend/blob/master/features/guidance/anticipate-lanes.feature#L317